File: Blind150/4.Stack/Go/daily_temperature.go
// https://leetcode.com/problems/daily-temperatures/description/
func dailyTemperatures(temperatures []int) []int {

    st := [][]int{}

    output := make([]int, len(temperatures))

    for pos, val := range(temperatures) {

        for len(st) > 0 && val > st[len(st)-1][0] {

            curTIdx := st[len(st)-1][1]
            st = st[:len(st)-1]
            output[curTIdx] = pos - curTIdx

        }

        st = append(st, []int{val, pos})
    }

    return output

}

File: Blind150/4.Stack/Python/daily_temperature.py
# https://leetcode.com/problems/daily-temperatures/description/
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:

        st = []

        output = [0]*len(temperatures)

        for pos, val in enumerate(temperatures):

            while st and val > st[-1][0]:

                cur_t_val, cur_t_idx = st.pop()
                output[cur_t_idx] = pos - cur_t_idx

            st.append((val, pos))

        return output
